docs/router/framework/solid/guide/custom-link.md

```diff
@@ -0,0 +1,25 @@
+---
+ref: docs/router/guide/custom-link.md
+replace: { 'react-router': 'solid-router' }
+---
+
+[//]: # 'BasicExampleImplementation'
+
+```tsx
+// TODO: Do the Solid equivalent of this.
```

Comment:
@birkskyum @brenelz could you please add in the Solid specific typings for this?

Reply:
It would look like this for solid:

```tsx
import { createLink, createRootRoute, Link, LinkComponent, Outlet } from "@tanstack/solid-router";
import { Component, JSX, Suspense } from "solid-js";

export const Route = createRootRoute({
  component: RootComponent
});


type BasicLinkProps = JSX.IntrinsicElements["a"] & {
  // Add any additional props you want to pass to the anchor element
};

const BasicLinkComponent: Component<BasicLinkProps> = (props) => (
  <a {...props} class="block px-3 py-2 text-red-700">
    {props.children}
  </a>
);


const CreatedLinkComponent = createLink(BasicLinkComponent)


export const CustomLink: LinkComponent<typeof BasicLinkComponent> = (props) => {
  return <CreatedLinkComponent preload={'intent'} {...props} />
}
```